modules/system/applications.nix

```diff
@@ -22,14 +22,26 @@ in
 
     system.activationScripts.applications.text = ''
       # Set up applications.
-      echo "setting up ~/Applications..." >&2
+      echo "setting up /Applications/Nix Apps..." >&2
+
+      ourLink () {
+        local link
+        link=$(readlink "$1")
+        [ -L "$1" ] && [ "''${link#*-}" = 'system-applications/Applications' ]
+      }
+
+      # Clean up for links created at the old location in HOME
+      if ourLink ~/Applications; then
+        rm ~/Applications
```

Comment:
Think you're missing a `mkdir ~/Applications` with the current user as the owner? i.e. after deleting the symlink there'll be no parent ~/Applications folder to symlink into. And we don't want to have a root owned one either

Reply:
No, it's not up to nix-darwin to create `~/Applications`. It's a directory that the user either creates for themselves or sometimes a program they run does it for them.
We don't need the directory anyway if we use `/Applications`. And since nix-darwin manages the system and all the users, it's unexpected behavior to create `~/Applications` just for the user running `darwin-rebuild`.

Reply:
Oh sorry, I've missed the fact you're using /Applications vs ~/Applications in this patch. That sounds good.